kahuna/public/js/components/gr-collections-panel/gr-collections-panel-node.html

```diff
@@ -11,39 +10,56 @@
 
         <button type="button"
                 class="node__toggler clickable"
-                ng:click="showChildren = !showChildren"
+                ng:click="ctrl.showChildren = !ctrl.showChildren"
                 ng:show="node.data.children.length > 0">
 
-            <gr-icon ng:show="showChildren">expand_more</gr-icon>
-            <gr-icon ng:hide="showChildren">chevron_right</gr-icon>
+            <gr-icon ng:show="ctrl.showChildren">expand_more</gr-icon>
+            <gr-icon ng:hide="ctrl.showChildren">chevron_right</gr-icon>
         </button>
 
         <a class="node__name flex-spacer"
-           ui:sref="search.results({query: (node.data.name)})">
+           ui:sref="search.results({query: ctrl.getCollectionQuery(node.data.content.pathId)})">
             {{node.data.name}}
         </a>
 
+
         <button class="node__action clickable" type="button"
                 ng:if="ctrl.deletable"
-                ng:click="ctrl.remove()">
+                ng:show="ctrl.editing"
+                ng:click="ctrl.remove()"
+                gr:tooltip="Delete this collection">
             <gr-icon-label gr-icon="delete"></gr-icon-label>
         </button>
 
         <button class="node__action clickable"
                 type="button"
-                ng:click="active = !active">
+                ng:show="ctrl.editing"
+                ng:click="active = !active"
+                gr:tooltip="Add a new sub-collection">
             <gr-icon>add_box</gr-icon>
         </button>
-    </div>
 
-    <form ng:show="active" ng:submit="ctrl.addChild(childName).then(clearForm);">
-        <input type="text" required ng:model="childName" />
-        <button type="submit">
-            <gr-icon-label gr-icon="check"></gr-icon-label>
-        </button>
-        <button type="button" ng:click="clearForm()" title="Close">
-            <gr-icon-label gr-icon="close"></gr-icon-label>
+        <button class="node__action clickable"
+                ng:if="!ctrl.editing && ctrl.hasImagesSelected && !ctrl.saving"
+                ng:click="ctrl.addImagesToCollection()"
+                gr:tooltip="Add selected images to this collection">
+            <gr-icon>add_to_photos</gr-icon>
         </button>
+        <span ng:if="ctrl.saving">Saving…</span>
+    </div>
+
+    <form class="node__add-child" ng:show="active" ng:submit="ctrl.addChild(childName).then(clearForm);">
```

Comment:
Would `ng:if` be more conservative in not having one form per tree node in the DOM?